mhl/hashlist.py

```diff
@@ -177,12 +177,19 @@ def log_hash_entry(self, hash_format):
             elif self.is_directory:
                 indicator = "d"
             hash_action = (hash_entry.action if hash_entry.action is not None else "").ljust(10)
-            logger.info("{0} {1}: {2} {3}: {4}".format(indicator,
-                                                       hash_entry.hash_format.rjust(6),
-                                                       hash_entry.hash_string.ljust(32),
-                                                       hash_action,
-                                                       self.path))
-
+            if hash_entry.structure_hash_string is not None:
+                logger.info("{0} {1}: {2} {3} (structure {4}): {5}".format(indicator,
+                                                           hash_entry.hash_format.rjust(6),
+                                                           hash_entry.hash_string.ljust(32),
+                                                           hash_entry.structure_hash_string,
+                                                           hash_action,
+                                                           self.path))
+            else:
+                logger.info("{0} {1}: {2} {3}: {4}".format(indicator,
+                                                           hash_entry.hash_format.rjust(6),
+                                                           hash_entry.hash_string.ljust(32),
+                                                           hash_action,
+                                                           self.path))
```

Comment:
```suggestion
            logger.info(f'{indicator} '
                        f'{hash_entry.hash_format.rjust(6)}: '
                        f'{hash_entry.hash_string.ljust(32)} '
                        f'( action: {hash_action} ) '
                        f'( structure: {hash_entry.structure_hash_string} ) '
                        f'{self.path}')`
```
using python f-strings we can eliminate this if / else. small suggestion for readability.
```

Reply:
Hello @jwaggs, I just tried your change. The purpose of the `if` clause is to avoid the `structure: ...` part in the output if the structure hash is `None`, but that was not solved with the proposed variant. Did you mean to switch the statement to something along the lines of `{structure or ""}` or how do you do that with f-strings? Generally I agree that getting rid of a large if statement that almost does the same in both cases definitively helps.

Reply:
Good point @ptrpfn - I've edited my comment to reflect your correction. You can either accept that snippet or commit it how you prefer.

Reply:
I changed that piece of log output to f-strings.